### PR TITLE
[Frontend] Integrate Ask Question Form With Backend

### PR DESCRIPTION
**Pull request by Aman16-ai**
## Issue that this pull request solves

 Closes: #783 

## Proposed changes

- integrating with backend

## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots
Screenshot (786)
Screenshot (787)

Please attach the screenshots of the changes made in case of change in user interface

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: frontend/src/pages/Q&A/Q&A.jsx [hunk 1]
```diff
@@ -93,27 +106,45 @@ function Ques(props) {
...
+          "content-type":"application/json"
+        },
+        body : JSON.stringify(formdata)
+      });
+      const data = await response.json();
+      setIsUploadingData(false)
+      setToastMessage("Q&A added successfully!")
+      setOpenToast(true)
+      setSeverity("success")
+      console.log(data)
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Q&A.jsx:125]: please remove all console.log statements

File: frontend/src/pages/Q&A/Q&A.jsx [hunk 2]
```diff
@@ -93,27 +106,45 @@ function Ques(props) {
...
-    if (Object.keys(errors).length !== 0) {
-      setFormErrors(errors);
-    }
-    if (Object.keys(errors).length !== 0) {
-      console.log(errors);
-    } else {
-      //Call the Server
-      console.log("Submitted");
-    }
+    console.log(formdata)
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Q&A.jsx:145]: please remove all console.log statements

File: frontend/src/pages/Q&A/Q&A.jsx [hunk 3]
```diff
@@ -93,27 +106,45 @@ function Ques(props) {
...
+      });
+      const data = await response.json();
+      setIsUploadingData(false)
+      setToastMessage("Q&A added successfully!")
+      setOpenToast(true)
+      setSeverity("success")
+      console.log(data)
+    }
+    catch(err) {
+      console.log(err)
```
Comment (Kajol-Kumari) [inline frontend/src/pages/Q&A/Q&A.jsx:128]: please remove it

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
@Aman16-ai Form validation isn't getting handled properly here. If any of the required field is empty, the api call shouldn't happen. But here I can see the success msg, although I got `400` in api response.

Ref - 
Screenshot 2022-04-13 at 9 38 51 AM

**Comment by Aman16-ai** [comment]
**fixed all validation bugs**
https://user-images.githubusercontent.com/74807466/163668459-767b6ad2-4174-405b-b7ed-e15be20aadce.mp4

**Review by Kajol-Kumari — APPROVED** [review]

**Pull request merged**